core/src/commonMain/kotlin/entity/Entitlement.kt

```diff
@@ -0,0 +1,90 @@
+package dev.kord.core.entity
+
+import dev.kord.common.entity.EntitlementType
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.Kord
+import dev.kord.core.behavior.EntitlementBehavior
+import dev.kord.core.behavior.GuildBehavior
+import dev.kord.core.behavior.UserBehavior
+import dev.kord.core.cache.data.EntitlementData
+import dev.kord.core.supplier.EntitySupplier
+import dev.kord.core.supplier.EntitySupplyStrategy
+import kotlinx.datetime.Instant
+
+public class Entitlement(
```

Comment:
should also override `equals` and `hashCode` based on its id

Reply:
Should I include application id?

Reply:
yes, good catch